lib/portlayer/exec/container_cache.go

```diff
@@ -79,3 +117,109 @@ func (conCache *containerCache) Remove(idOrRef string) {
 	}
 
 }
+
+func isContainerID(id string) bool {
+	// ID should only be lower case chars & numbers
+	match, _ := regexp.Match("^[a-z0-9]*$", []byte(id))
+	return match
+}
+
+// call back for the event stream
+func callme(ie event.Event, sess *session.Session) {
+	// grab the container from the cache
+	container := containers.Container(ie.Reference())
+	if container != nil {
+
+		newState := eventedState(ie.Event(), container.State)
+		// do we have a state change
+		if newState != container.State {
+			switch newState {
+			case StateStopping:
+				// set state only in cache -- allow tether / tools to update vmx
+				log.Debugf("Container(%s) Shutdown via OOB activity", container.ExecConfig.ID)
+				container.State = StateStopped
+			case StateRunning:
+				// set state only in cache -- allow tether / tools to update vmx
+				log.Debugf("Container(%s) started via OOB activity", container.ExecConfig.ID)
+				container.State = newState
+			case StateStopped:
+				log.Debugf("Container(%s) stopped via OOB activity", container.ExecConfig.ID)
+				container.State = newState
+				// a powered off event will by pass guest shutdown, so we need to update
+				// the exit status
+				containerUpdate(sess, container, "", -1)
+			case StateSuspended:
+				//set state only in cache
+				log.Debugf("Container(%s) suspened via OOB activity", container.ExecConfig.ID)
+				container.State = newState
+			case StateRemoved:
+				log.Debugf("Container(%s) removed via OOB activity", container.ExecConfig.ID)
+				// remove from cache
+				containers.Remove(container.ExecConfig.ID)
+
+			}
+		}
+	} else {
+		// TODO: do we need to worry about this?  any further evaluation, etc?
+		log.Warnf("ContainerCache miss for VM(%s) in VCH Resource Pool: Event(%s)", ie.Reference(), ie.Event())
+	}
+	return
+}
+
+// best effort container udpate -- if there's an issue we'll warn and return
+func containerUpdate(sess *session.Session, container *Container, start string, exitStatus int) {
+	// first update exec from vsphere
+	ctx, cancel := context.WithTimeout(context.TODO(), propertyCollectorTimeout)
+	defer cancel()
```

Comment:
Not sure I understand the purpose of this. This will never timeout and no one calls cancel so why we bother to create?